client/src/components/messenger/Conversations.js

```diff
@@ -1,53 +1,112 @@
+    const token = useSelector((state) => state.user.token);
+
+    useEffect(() => {
+        // if this is the first time rendering, get user conversations from server
+        if (!init) {
+            (async () => setConversations(await ConversationsController.getConversations(token)))();
+            setInit(true);
+        }
+    }, [init]);
+
+    const handleClickAdd = () => setDialogOpen(true);
+    const addConversation = (participants) => {
+        setDialogOpen(false); // close add dialog
+        // ask the server to create a new conversation with the list participants
+        if (participants)
+            (async () => {
+                let result = ConversationsController.createConversation(token, participants);
+                if (result.success) { // add new conversation to the back of the list
+                    let { conversation } = result;
+                    setConversations(conversations.length > 0 ? [...conversations, conversation]: [conversation]);
+                    // auto open chat after creation
+                    selectConversation(conversation.id, conversation.title);
+                }
+                else // display error message
+                    alert(result.msg);
+            })();
+    };
 
-let conversations = [
-    {
-        name: "John",
-        img: "https://picsum.photos/200?random=1",
-        lastMsg: "Whats up",
-        lastDate: "11:00 AM",
-        unread: 1,
-    }, {
-        name: "Jenny",
-        img: "https://picsum.photos/200?random=2",
-        lastMsg: "I missed you",
-        lastDate: "5:00 PM",
-        unread: 0,
-    }, {
-        name: "Unknown",
-        img: "https://picsum.photos/200?random=3",
-        lastMsg: "Lorem ipsum dolor sit amet.",
-        lastDate: "Yesterday",
-        unread: 12,
-    }, {
-        name: "Josh",
-        img: "https://picsum.photos/200?random=4",
-        lastMsg: "Lorem ipsum sit dolor amet.",
-        lastDate: "8/28/21",
-        unread: 4,
-    }
-];
-
-const Conversations = () => {
     return (
         <div id="chat_container">
-            <h1>Conversations</h1>
+            <div className="main_panel_header">
+                <h1>Conversations</h1>
+                <button className="options" title="options" onClick={() => handleClickAdd()}>
+                    <img src={plus} alt="options"/>
+                </button>
+                <AddConversationDialog open={dialogOpen} onClose={addConversation} token={token} />
+            </div>
             <div id="chats">
                 {conversations.map((chat, i) =>
                     <ConversationItem
-                        name={chat.name}
-                        img={chat.img}
+                        name={chat.title}
+                        img={chat.imgUrl}
                         lastMsg={chat.lastMsg}
-                        lastDate={chat.lastDate}
+                        lastDate={ConversationsController.evaluateDate(chat.lastDate)}
```

Comment:
Where does lastDate come from?

Reply:
I believe this is the creation date of the last message in that conversation

Reply:
Yh I get that but where is it assigned? Same with lastMsg